api/src/routers/meals.js

```diff
@@ -0,0 +1,92 @@
+import express from "express";
+import { StatusCodes } from "http-status-codes";
+import db from "../db.js";
+
+const router = express.Router();
+//GET api/meals route
+//maxPrice	Number	Returns all meals that are cheaper than maxPrice.	api/meals?maxPrice=90
+router.get("/", async (req, res) => {
+  try {
+    let query = db("meal").select("*");
+    if (req.query.maxPrice) {
+      const maxPrice = parseFloat(req.query.maxPrice);
+      if (isNaN(maxPrice)) {
+        return res.status(400).json({ error: "Invalid maxPrice parameter" });
+      }
+      query = query.where("price", "<=", maxPrice);
+    }
+
+    // availableReservations	Boolean	Returns all meals that still have available spots left, if true. If false, return meals that have no available spots left.1	api/meals?availableReservations=true
+
+    if (req.query.availableReservations) {
+      const available =
+        req.query.availableReservations.toLowerCase() === "true";
+      if (available) {
+        query = query.where(
+          "max_reservations",
+          ">",
+          db.raw("reservations_count")
```

Comment:
```suggestion
          db.ref("reservations_count")
```
ref is safer and better than raw